models/DataObject/ClassDefinition.php

```diff
@@ -1422,6 +1422,15 @@ public function getCompositeIndices(): array
      */
     public function setCompositeIndices($compositeIndices)
     {
+        $class = $this->getFieldDefinitions([]);
+        foreach($compositeIndices as $indexInd => $compositeIndex){
+            foreach($compositeIndex["index_columns"] as $fieldInd => $fieldName){
+                if(isset($class[$fieldName]) && str_contains(strtoupper($class[$fieldName]->getFieldType()), "MANYTOONE")){
```

Comment:
I would suggest writing it like this, than we aren't hard coding a field type name.
```suggestion
                if(isset($class[$fieldName]) && $class[$fieldName] instanceof ManyToOneRelation){
```

Reply:
I would suggest writing it like this, than we aren't hard coding a field type name.

Reply:
> I like this pattern. One comment I have is that once saved, we are able to remove one of these fields, and than re-add the ManyToOneRelation field which makes it throw an error. ![image](https://user-images.githubusercontent.com/58705994/208171326-c689171b-0768-45dc-b99d-65945361bcf1.png)

I would think doing a array_unique check on the back end would fix this issue but im, getting different issues after doing that. The backend seems to like a certain order for the indicies that seems random to me.